src/server/pkg/deploy/cmds/cmds.go

```diff
@@ -75,7 +75,11 @@ func DeployCmd(noMetrics *bool) *cobra.Command {
 			if err := assets.WriteLocalAssets(manifest, opts, hostPath); err != nil {
 				return err
 			}
-			return maybeKcCreate(dryRun, manifest)
+			ret := maybeKcCreate(dryRun, manifest)
```

Comment:
Why are we creating this variable here and returning it later?

Also now that I think about it, maybe we can make the following code part of `maybeKcCreate`, so that we don't have to have duplicated code in 4 different places.  You just need to pass `opts` as an argument to `maybeKcCreate`.  Lemme know if that doesn't make sense.